plugin/src/test/groovy/com/novoda/buildproperties/internal/FilePropertiesEntriesTest.groovy

```diff
@@ -44,12 +47,12 @@ class FilePropertiesEntriesTest {
     }
 
     @Test
-    void shouldThrowIllegalArgumentExceptionWhenTryingToAccessValueOfUndefinedProperty() {
+    void shouldThrowExceptionWhenTryingToAccessValueOfUndefinedProperty() {
         try {
             entries['notThere'].string
-            fail('IllegalArgumentException expected')
-        } catch (IllegalArgumentException e) {
-            assertThat(e.getMessage()).startsWith("No value defined for property 'notThere'")
+            fail('Exception expected')
+        } catch (Exception e) {
+            assertThat(e.getMessage()).contains("Unable to find value for key 'notThere'")
```

Comment:
This might be a silly question. Can we not use the thrown junit rule instead of having try catches everywhere? You could then use matchers on the thrown rule.